src/negative_tests.rs

```diff
@@ -321,3 +385,99 @@ fn test_iterator_variable_redefinition() {
         ErrorKind::DuplicateDefinition(..)
     ));
 }
+
+#[test]
+fn test_boolean_and_fail() {
+    let code = r#"
+    fn thing(xx: Field, yy: Bool) {
+        let zz = xx && yy;
+    }
+    "#;
+
+    let res = tast_pass(code).0;
+    assert!(matches!(res.unwrap_err().kind, ErrorKind::MismatchType(..)));
+}
+
+#[test]
+fn test_boolean_or_fail() {
+    let code = r#"
+    fn thing(xx: Field, yy: Bool) {
+        let zz = xx || yy;
+    }
+    "#;
+
+    let res = tast_pass(code).0;
+    assert!(matches!(res.unwrap_err().kind, ErrorKind::MismatchType(..)));
+}
+
+#[test]
+fn test_addition_mismatch() {
+    let code = r#"
+    fn thing(xx: Field) -> Field {
+        let yy = xx + 1;
+        return yy + true;
+    }
+    "#;
+
+    let res = tast_pass(code).0;
+    assert!(matches!(res.unwrap_err().kind, ErrorKind::MismatchType(..)));
+}
+
+#[test]
+fn test_multiplication_mismatch() {
+    let code = r#"
+    fn thing(xx: Field) -> Field {
+        let yy = xx * 2;
+        return yy + true;
+    }
+    "#;
+
+    let res = tast_pass(code).0;
+    assert!(matches!(res.unwrap_err().kind, ErrorKind::MismatchType(..)));
+}
+
+#[test]
+fn test_asm_snapshot_mismatch_fail() {
```

Comment:
The asm will be compared in the `examples.rs`. It is more a higher level testing category.

The test cases here are more about testing the expected exceptions, at lower level.